### PR TITLE
add Memtargetgettype

### PR DESCRIPTION
**Pull request by lplewa**
depends on #631

### CODE REVIEW

File: include/umf/memtarget.h [hunk 1]
```diff
@@ -10,13 +10,27 @@
...
+typedef enum umf_memtarget_type_t {
+    UMF_MEMTARGET_TYPE_UNKNOWN = 0,
+    UMF_MEMTARGET_TYPE_NUMA = 1,
+} umf_memtarget_type_t;
+
+/// \brief Gets the type of the memory target.
+/// \param hMemtarget handle to the memory target
+/// \param type [out] type of the memory target
+/// \return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+umf_result_t umfMemtargetGetType(umf_const_memtarget_handle_t hMemtarget,
```
[the remark this answers is not in the document]
Comment (lplewa) [reply include/umf/memtarget.h:31]: Done

File: include/umf/memtarget.h [hunk 2]
```diff
@@ -10,13 +10,27 @@
 #ifndef UMF_MEMTARGET_H
 #define UMF_MEMTARGET_H 1
 
+#include "umf/base.h"
```
[the remark this answers is not in the document]
Comment (lplewa) [reply include/umf/memtarget.h:13]: Done

File: test/memspaces/memspace_numa.cpp [hunk 3]
```diff
@@ -6,6 +6,7 @@
 #include "memspace_fixtures.hpp"
 #include "memspace_helpers.hpp"
 #include "memspace_internal.h"
+#include "umf/memspace.h"
```
[the remark this answers is not in the document]
Comment (lplewa) [reply test/memspaces/memspace_numa.cpp:9]: done

File: src/memtargets/memtarget_numa.c [hunk 4]
```diff
@@ -19,6 +19,7 @@
 #include "mempolicy_internal.h"
 #include "memtarget_numa.h"
 #include "topology.h"
+#include "umf/base.h"
```
[the remark this answers is not in the document]
Comment (lplewa) [reply src/memtargets/memtarget_numa.c:22]: Done

**Review by bratpiorka — APPROVED** [review]

**Comment by PatKamin** [comment]
Please, squash latest two commits

**Review by PatKamin — APPROVED** [review]

**Pull request merged**